google_bigquery_writer/writer.py

```diff
@@ -196,3 +209,30 @@ def write_table_sync(
                     job.errors
                 )
                 raise UserException(message)
+
+    def _split_csv(self, csv_file, table_name, nr_of_slices):
+        try:
+            result = subprocess.run(
+                [
+                    '/home/cli_linux_amd64',
+                    '--table-name=' + str(table_name),
+                    '--table-input-path=' + csv_file,
+                    '--mode=slices',
+                    '--number-of-slices=' + str(nr_of_slices),
+                    '--table-output-path=' + self.TEMP_PATH,
+                    '--table-output-manifest-path=/home/data/dump.manifest',
+                    '--gzip=false'
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE
+            )
+
+            if result.returncode != 0:
+                raise UserException(f"Error running cli_linux_amd64: {result.stderr.strip()}")
+
+        except Exception as e:
+            raise UserException(f"Error running cli_linux_amd64: {e}")
```

Comment:
I would keep that as App error. The user has probably no way of fixing it so we should know about it.